pom.xml

```diff
@@ -211,6 +211,9 @@ under the License.
                     <groupId>org.apache.avro</groupId>
                     <artifactId>avro-maven-plugin</artifactId>
                     <version>${avro.version}</version>
+                    <configuration>
+                      <stringType>String</stringType>
```

Comment:
There were two reason why I wanted to use `String`.
1) The classes in `org.apache.reef.tang.implementation.types` such as `ClassNodeImpl` take `String` as arguments. When we deserialize a ClassHierarchy, we should pass the `CharSequence` to `String` type. In this case, `toString()` method is used all the time, which drops readability.
2) Moreover, since some string field can be null, we should make sure those fields are not null before calling `toString()` which could throw `NullPointerException` when a field is empty.
I did not notice that this potentially invite a performance issue. I will change to stick to the default `CharSequence`.